hoot-services/src/main/java/hoot/services/utils/DbUtils.java

```diff
@@ -379,14 +380,20 @@ public static void deleteEmptyFolders() {
      * Sets the parent directory for the specified folder
      *
      * @param folderId folder id whos parent we are setting
-     * @param parentId parent directory id that the folder will get linked to
+     * @param parent parent folder that the folder will get linked to
      */
-    public static void setFolderParent(Long folderId, Long parentId) {
-        createQuery()
+    public static void setFolderParent(Long folderId, Folders parent) {
+        SQLUpdateClause query = createQuery()
                 .update(folders)
                 .where(folders.id.eq(folderId))
-                .set(folders.parentId, parentId)
-                .execute();
+                .set(folders.parentId, parent.getId());
+
+        // dont want a private folder moved to root to become public
```

Comment:
let's add a comment like
`// but otherwise inherit the visibility of the new parent folder`